### PR TITLE
Parse AV1 OBU in case dependency descriptor is not present for resolution stats

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: src/DependencyDescriptorLayerSelector.cpp [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include <vector>
 
 #include "DependencyDescriptorLayerSelector.h"
+#include "av1/AV1.h"
```
[the remark this answers is not in the document]
Comment (harryz2000) [reply src/DependencyDescriptorLayerSelector.cpp:4]: I tried to use conditions in the class for AV1. But yes, using a separate class would be cleaner.

File: src/av1/AV1LayerSelector.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	uint32_t obuSize = reader.GetLeft();
+	if (header->W != 1)
+	{
+		obuSize = reader.DecodeLev128();
+	}
+	
+	// Not enough data
+	if (obuSize > reader.GetLeft()) return infos;
+	
+	auto info = GetObuInfo(reader.PeekData(), obuSize);
```
Comment (murillo128) [inline src/av1/AV1LayerSelector.cpp:40]: we would need to refactor this into ObuInfo::Parse to be consistent with the rest of the objects

Comment (murillo128) [reply src/av1/AV1LayerSelector.cpp:40]: In the refactor, we should pass the reader, so we don't have to store the `payload` pointer on the `ObuInfo`

**Review by murillo128 — CHANGES_REQUESTED** [review]

**Comment by murillo128** [comment]
lgtm, we can merge and do the refactor in an upcoming PR

**Pull request merged**